pkg/util/kubernetes/kubelet/kubelet_client.go

```diff
@@ -198,6 +213,19 @@ func getKubeletClient(ctx context.Context) (*kubeletClient, error) {
 		tokenPath:      kubeletTokenPath,
 	}
 
+	if kubeletUseAPIServer {
+		apiServerHost := os.Getenv("KUBERNETES_SERVICE_HOST")
+		apiServerPort := os.Getenv("KUBERNETES_SERVICE_PORT")
+		if apiServerHost == "" || apiServerPort == "" {
+			return nil, fmt.Errorf("failed to determine API server host/port")
+		}
+
+		clientConfig.useAPIServer = true
+		clientConfig.apiServerHost = fmt.Sprintf("https://%s:%s", apiServerHost, apiServerPort)
+
+		log.Infof("kubeletUseApiServer set to true, pod list queries will be sent to the apiserver at: %s/api/v1/pods", clientConfig.apiServerHost)
```

Comment:
nit: this only impacts a small number of users but wondering if `Debug` is a more appropriate level since `useAPIServer` would be explicitly set (or if we plan to set that for GKE users automatically, maybe removing mention of `kubeletUseApiServer` would be less confusing?)